apps/web/src/lib/api/external/laskugeneraattori/actions.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  type InvoiceGeneratorFormState,
+  type LaskugeneraattoriRequest,
+} from "./index";
+
+// https://stackoverflow.com/a/22015930
+const zip = <T>(...arr: T[][]) =>
+  Array(Math.max(...arr.map((a) => a.length)))
+    .fill(0)
+    .map((_, i) => arr.map((a) => a[i]));
+
+export const laskugeneraattoriBaseUrl =
+  process.env.NEXT_PUBLIC_LASKUGENERAATTORI_URL ??
+  "https://laskutus.tietokilta.fi";
+
+export async function SaveAction(
+  currentState: unknown,
+  formData: FormData,
+): Promise<InvoiceGeneratorFormState> {
+  "use client";
```

Comment:
I don't think "use client" belongs here :eyes: All form actions should be marked with "use server" either first row of the function or top of the file

Reply:
The idea here was to send the request to laskugeneraattori from the client side, so I put "use client" there. Seemed to work fine on my machine, but maybe using these form actions is not the correct way to do it ¯\_(ツ)_/¯

I'll change it to "use server" atleast for now, if that works better

Reply:
Form actions are by default server only, if you were to do it client side you could just plug in the regular onSubmit on the form, but then you probably can't use the useFormAction etc. hooks

I think it's better we do it server-side, since then we can have support for sending invoices without JavaScript enabled. I think we have some dedicated tinfoil-hats that prefer to keep everything as tight as possible 😅  and it was one of the first requests for the new ilmomasiina to support sign up without JavaScript

Reply:
"use client" is used to mark components as client components instead of RSC, I don't think it has any effect on functions necessarily. The function would already be a client function, since it is imported into a client component (the form as "use client" at the top), without adding the "use server"